src/acomms/modemdriver/popoto_driver.cpp

```diff
@@ -9,116 +33,110 @@
     int payload_mode = popoto_driver_cfg().payload_mode();
     int start_timeout = popoto_driver_cfg().start_timeout();
-  
-    // Set the default baud 
+
+    // Set the default baud
     if (!driver_cfg_.has_serial_baud())
         driver_cfg_.set_serial_baud(DEFAULT_BAUD);
 
     glog.is(DEBUG1) && glog << group(glog_out_group()) << "PopotoDriver: Starting modem..."
                             << std::endl;
     ModemDriverBase::modem_start(driver_cfg_);
-    
-    
+
     // -------------------------- set the Popoto CFG params in the right format ------------
     std::stringstream raw;
     raw << "setvaluef TxPowerWatts " + std::to_string(modem_p) << "\n";
     signal_and_write(raw.str());
-    
+
     raw.str(""); // clear the string stream
     raw << "setvaluei PayloadMode " + std::to_string(payload_mode) << "\n";
     signal_and_write(raw.str());
-    
-    raw.str(""); 
+
+    raw.str("");
     raw << "setvaluei LedEnable " + std::to_string(0) << "\n";
     signal_and_write(raw.str());
 
-    raw.str(""); 
+    raw.str("");
     raw << "setvaluei LocalID " << driver_cfg_.modem_id() << "\n";
     signal_and_write(raw.str());
-    
+
     // Poll the modem temp and battery voltage
     signal_and_write("getvaluef BatteryVoltage\n");
     signal_and_write("getvaluef Temp_Ambient\n");
 
     // Check if modem has started
     std::string in;
     int startup_elapsed_ms = 0;
-    
+
     while (!modem_read(&in))
-    {   
+    {
         usleep(100000); // 100 ms
         startup_elapsed_ms += 100;
 
-        if (startup_elapsed_ms/1000 >= start_timeout)
+        if (startup_elapsed_ms / 1000 >= start_timeout)
             throw(ModemDriverException("Modem physical connection failed to startup.",
                                        protobuf::ModemDriverStatus::STARTUP_FAILED));
     }
 
-    std::cout << "Modem " << driver_cfg_.modem_id()  << " initialized OK." << std::endl;
+    glog.is(DEBUG1) && glog << "Modem " << driver_cfg_.modem_id() << " initialized OK."
+                            << std::endl;
 }
 
-void goby::acomms::popotoDriver::shutdown()
-{  
-
-    ModemDriverBase::modem_close();
-}
+void goby::acomms::PopotoDriver::shutdown() { ModemDriverBase::modem_close(); }
 
 // --------------------------- Outgoing msgs ------------------------------------------------
-void goby::acomms::popotoDriver::handle_initiate_transmission(
+void goby::acomms::PopotoDriver::handle_initiate_transmission(
     const protobuf::ModemTransmission& orig_msg)
 {
     protobuf::ModemTransmission msg = orig_msg;
     // Poll the modem temp and battery voltage before each transmission
     signal_and_write("getvaluef BatteryVoltage\n");
     signal_and_write("getvaluef Temp_Ambient\n");
-    
+
     switch (msg.type())
     {
         case protobuf::ModemTransmission::DATA:
         {
-            //msg.set_max_frame_bytes(1024); // Might not need to set this
+            msg.set_max_num_frames(1);
+
+            if (!msg.has_max_frame_bytes())
```

Comment:
If the caller of handle_initiate_transmission (e.g. MACManager) doesn't have a max_frame_bytes set, choose a sane default so that Queue doesn't fill this packet up indefinitely.